integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -240,9 +242,13 @@ public int chown(String path, @uid_t long uid, @gid_t long gid) {
   public int create(String path, @mode_t long mode, FuseFileInfo fi) {
     final AlluxioURI uri = mPathResolverCache.getUnchecked(path);
     final int flags = fi.flags.get();
+    final String name = uri.getName();
     LOG.trace("create({}, {}) [Alluxio: {}]", path, Integer.toHexString(flags), uri);
 
     try {
+      if (name.length() > MAX_NAME_LENGTH) {
+        throw new NameTooLongException(name + " beyond " + MAX_NAME_LENGTH + " characters");
```

Comment:
instead of throwing and catch the exception right away, can we just `LOG.error` and `return -ErrorCodes.ENAMETOOLONG()`? Also move this check outside the `try` block

Reply:
modified.